app.js

```diff
@@ -271,15 +283,32 @@ const completeRecoveryValidation = async ctx => {
 
     const account = await models.Account.findOne({ where: { accountId } });
 
+    if (!account) {
+        ctx.throw(401, 'account does not exist');
+    }
+
     const [recoveryMethod] = await account.getRecoveryMethods({ where: {
         kind: method.kind,
         detail: method.detail,
         securityCode: securityCode
```

Comment:
@mattlockyer looks like I can just avoid supplying `securityCode` to bypass this check.

Reply:
Looks like it doesn't work in practice because `sequelize` doesn't treat `undefined` value as absent key here. 

However I don't think it's safe to rely on this implicit behavior. We should have an explicit test for no security code supplied. I believe we discussed this before.

Reply:
added: `if (!securityCode || isNaN(parseInt(securityCode, 10)) || securityCode.length !== 6)`

Reply:
@mattlockyer this needs test (similar as wrong code).